sub/dec_sub.c

```diff
@@ -119,6 +123,18 @@ static void wakeup_demux(void *ctx)
     mp_dispatch_interrupt(q);
 }
 
+static void sub_destroy_cached_pkts(struct dec_sub *sub)
+{
+    while (sub->num_cached_pkts) {
+        struct demux_packet *pkt = sub->cached_pkts[sub->num_cached_pkts - 1];
+        if (pkt)
+            talloc_free(pkt);
+        MP_TARRAY_REMOVE_AT(sub->cached_pkts, sub->num_cached_pkts,
```

Comment:
Since you remove all elements, not really needed to do remove at, just iterate all of them in loop and set num to zero after.